accounts.go

```diff
@@ -70,7 +71,37 @@ func (h *accountsHandler) Get(c *gin.Context) {
 }
 
 func (h *accountsHandler) List(c *gin.Context) {
-	c.JSON(http.StatusOK, gin.H{"message": "unimplemented"})
+	bearer, err := h.authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	offset, err := strconv.ParseInt(c.Query("offset"), 10, 64)
+	if err != nil {
+		writeError(c, http.StatusBadRequest, err)
+		panic(err)
+	}
+
+	limit, err := strconv.ParseInt(c.Query("limit"), 10, 64)
+	if err != nil {
+		writeError(c, http.StatusBadRequest, err)
+		panic(err)
+	}
```

Comment:
You should never `panic()` in this context. The uses for `panic()` are pretty limited. Use it when the program should exit immediately because it cannot recover from failure. Panics are not like exceptions and terminate the program. In our case, we have a panic-catching middleware which prevents the server from crashing but `panic()` is still inappropriate in this scenario.
```suggestion
	offset, err := strconv.ParseInt(c.Query("offset"), 10, 64)
	if err != nil {
		writeError(c, http.StatusBadRequest, err)
		return
	}

	limit, err := strconv.ParseInt(c.Query("limit"), 10, 64)
	if err != nil {
		writeError(c, http.StatusBadRequest, err)
		return
	}
```